internal/db/role_test.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2024 Canonical.
+
+package db_test
+
+import (
+	"context"
+	"fmt"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/google/uuid"
+
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+)
+
+func (s *dbSuite) TestAddRole(c *qt.C) {
+	ctx := context.Background()
+
+	uuid := uuid.NewString()
+	c.Patch(db.NewUUID, func() string {
+		return uuid
+	})
+
+	_, err := s.Database.AddRole(ctx, "test-role")
+	c.Check(errors.ErrorCode(err), qt.Equals, errors.CodeUpgradeInProgress)
+
+	err = s.Database.Migrate(context.Background(), false)
+	c.Assert(err, qt.IsNil)
+
+	roleEntry, err := s.Database.AddRole(ctx, "test-role")
+	c.Assert(err, qt.IsNil)
+	c.Assert(roleEntry.UUID, qt.Not(qt.Equals), "")
+
+	_, err = s.Database.AddRole(ctx, "test-role")
+	c.Assert(errors.ErrorCode(err), qt.Equals, errors.CodeAlreadyExists)
+
+	re := dbmodel.RoleEntry{
+		Name: "test-role",
+	}
+	tx := s.Database.DB.First(&re)
+	c.Assert(tx.Error, qt.IsNil)
+	c.Assert(re.ID, qt.Equals, uint(1))
+	c.Assert(re.Name, qt.Equals, "test-role")
+	c.Assert(re.UUID, qt.Equals, uuid)
+}
+
+func (s *dbSuite) TestGetRole(c *qt.C) {
+	uuid1 := uuid.NewString()
+	c.Patch(db.NewUUID, func() string {
+		return uuid1
+	})
+
+	err := s.Database.GetRole(context.Background(), &dbmodel.RoleEntry{})
+	c.Check(errors.ErrorCode(err), qt.Equals, errors.CodeUpgradeInProgress)
+
+	err = s.Database.Migrate(context.Background(), false)
+	c.Assert(err, qt.IsNil)
+
+	role := &dbmodel.RoleEntry{}
+	err = s.Database.GetRole(context.Background(), role)
+	c.Check(errors.ErrorCode(err), qt.Equals, errors.CodeNotFound)
```

Comment:
hmm.. but if you add a role above this, then we expect to get one role back - whichever is first by some magic order.

Reply:
Would you like me to break it into its own test? If I do, it'd probably be worth breaking up all cases.  But many of our tests bundle many cases into one so it'd be a bit of an outlier